File: sprint4/By languages/go/lesson1code4.go
type Pair struct {
    key string
    value interface{}
}

type Map struct {
    pairs []*Pair
}

func (m *Map) Get(key string) interface{} {
    for _, pair := range m.pairs {
        if pair.key == key {
            return pair.value
        }
    }
    return nil
}

func (m *Map) Set(key string, value interface{}) {
    for _, pair := range m.pairs {
        if pair.key == key {
            pair.value = value
            return
        }
    }
    m.pairs = append(m.pairs, &Pair{key: key, value: value})
}

File: sprint4/By languages/kotlin/lesson1code4.kt
class Map {
    var pairs = arrayListOf<Pair>()

    fun get(key: Any): Any? {
        for (pair in pairs) {
            if (pair.key == key) {
                return pair.value
            }
        }
        return null
    }

    fun set(key: Any, value: Any) {
        for (pair in pairs) {
            if (pair.key == key) {
                pair.value = value
                return
            }
        }
        pairs.add(Pair(key, value))
    }
}

data class Pair(var key: Any, var value: Any)
